File: classroom-booking/src/main/java/com/ilerna/classroom_booking/service/AvailabilityService.java
package com.ilerna.classroom_booking.service;

import java.util.List;

import com.ilerna.classroom_booking.model.Availability;

public interface AvailabilityService {
    Availability saveAvailability(Availability availability);
    Availability getAvailabilityById(Long id);
    List<Availability> getAllAvailabilities();
    Availability updateAvailability(Availability availability);
    void deleteAvailabilityById(Long id);
}

File: classroom-booking-frontend/src/app/services/availability.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Availability } from '../models/availability';

@Injectable({
  providedIn: 'root'
})
export class AvailabilityService {
  private baseUrl = 'http://localhost:8080/availability';

  constructor(private http: HttpClient) { }

  getAllAvailabilities(): Observable<Availability[]> {
    return this.http.get<Availability[]>(this.baseUrl);
  }

  getAvailabilityById(id: number): Observable<Availability> {
    return this.http.get<Availability>(`${this.baseUrl}/${id}`);
  }

  saveAvailability(availability: Availability): Observable<Availability> {
    return this.http.post<Availability>(this.baseUrl, availability);
  }

  updateAvailability(availability: Availability): Observable<Availability> {
    return this.http.put<Availability>(`${this.baseUrl}/${availability.id}`, availability);
  }

  deleteAvailabilityById(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }
}
